base-notebook/Dockerfile

```diff
@@ -1,10 +1,8 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-# Ubuntu 22.04 (jammy)
-# https://hub.docker.com/_/ubuntu/tags?page=1&name=jammy
-ARG ROOT_CONTAINER=ubuntu:22.04
-
+ARG OWNER=jupyter
+ARG ROOT_CONTAINER=$OWNER/base-jupyter
```

Comment:
Let's have BASE_CONTAINER, as in other inherited images.

Reply:
Sounds good.  I brought up the backward compatibility topic because anyone building from `base-notebook` today that wishes to use a different base container has something like the following in their build script:
```
docker build --build-arg ROOT_CONTAINER=my-base-container:tag -t jupyter/base-notebook:dev .
```
and those instances will no longer produce the expected result.  If that isn't a supported use-case, then I'm cool with only having `BASE_CONTAINER`.

One compromise might be something like this:
```
ARG OWNER=jupyter
ARG ROOT_CONTAINER
ARG BASE_CONTAINER=${ROOT_CONTAINER:-${OWNER}/base-jupyter}
```

Reply:
I was also taking the same argument with the `NB_` arguments.   Existing scripts against `base-notebook` could be using `--build-arg NB_USER=alice`.

Reply:
My latest push includes the b/c approaches.  I'm happy to remove them if you don't feel that's something to support.

Reply:
I think it's better to remove your b/c approach.
With your approach, it's still not b/c compatible.
Because anyone who will try to build base-notebook, which is not derived from jupyter-base will encounter so many bugs.
Yes, we make everyone build `jupyter-base` and then `base-notebook` (instead of only building `minimal-notebook` as before), but that's the price we have to pay. And I honestly believe that's ok.